tower/src/layer.rs

```diff
@@ -14,12 +14,12 @@ pub trait LayerExt<S, Request>: Layer<S, Request> {
     /// `middleware` to services being wrapped.
     ///
     /// This defines a middleware stack.
-    fn chain<T>(self, middleware: T) -> util::Chain<Self, T>
+    fn chain<T>(self, middleware: T) -> util::Stack<Self, T>
```

Comment:
It's possible we've already discussed this, but I feel like the type and method ought to share the same name?
```suggestion
    fn stack<T>(self, middleware: T) -> util::Stack<Self, T>
```

Reply:
👍 on this

Reply:
Ah yes... but I was thinking about getting rid of that fn. I don't think it is needed w/ `ServiceBuilder`.

Reply:
Thanks for catching this though.

Reply:
I still think it could be useful 🤷‍♂️ I don't see why there is a reason to remove it over keeping it.

Reply:
Can always bring it back later. I rather stay on the conservative side initially.

Reply:
There is only 1 fn right now... `Stack::new` can be called directly, and `ServiceBuilder` handles the use case too.